contracts/modules/cover/Cover.sol

```diff
@@ -126,7 +169,7 @@ contract Cover is ICover, MasterAwareV2 {
         uint96(premiumInAsset)
       );
 
-    coverNFT.safeMint(owner, coverId);
+    ICoverNFT(coverNFT).safeMint(owner, coverId);
```

Comment:
Storing it as an address allows you to use Cover.sol in contracts which use older solidity versions which do not allow non-primitive return values such as interfaces or structs without using experimental features. At the same time it could very well be stored in contract bytecode since coverNFT will be immutable.

Reply:
is there a use case for this in our codebase? reading coverNFT from legacy code